lib/services/local/access_list.go

```diff
@@ -1029,3 +1043,18 @@ func (a *AccessListService) VerifyAccessListCreateLimit(ctx context.Context, tar
 	const limitReachedMessage = "cluster has reached its limit for creating access lists, please contact the cluster administrator"
 	return trace.AccessDenied(limitReachedMessage)
 }
+
+// conditionallyPreserveAWSIdentityCenterLabels preserves member labels if
+// it originated from AWS Identity Center plugin.
+// The Web UI does not preserve metadata labels so this function should be called
+// in every update/upsert member calls.
```

Comment:
The comment should explain _why_ we need to preserve them. Tbh it's still not super clear to me. What scenario is this needed for?

Reply:
> The Web UI does not preserve metadata labels so this function should be called
> in every update/upsert member calls.

That is meant to explain the _why_ but I can see it failed to convey the message. 

See this issue I created yesterday https://github.com/gravitational/teleport.e/issues/5415. Right now when members are added/removed from the access list using the web UI, we do not preserve existing labels and metadata name. Its a full replacement of old members with new member struct. That calls for preserving existing labels otherwise it will be updated to be nil since labels are not handled at all https://github.com/gravitational/teleport.e/blob/master/lib/web/accesslist.go#L250C1-L269C2

Reply:
Any suggestion on the commentary? I am falling short on how to explain that otherwise. I put ticket link there for reference.